src/components/AttachmentPicker/index.native.js

```diff
@@ -157,29 +156,20 @@ class AttachmentPicker extends Component {
             },
         ];
 
-        this.setResult = this.setResult.bind(this);
         this.close = this.close.bind(this);
-        this.completeAttachmentSelection = this.completeAttachmentSelection.bind(this);
+        this.pickAttachment = this.pickAttachment.bind(this);
     }
 
     /**
-     * Store the selected attachment mapped to an appropriate file interface
+     * Handles the image/document picker result and
+     * sends the selected attachment to the caller (parent component)
      *
      * @param {ImagePickerResponse|DocumentPickerResponse} attachment
      */
-    setResult(attachment) {
+    pickAttachment(attachment) {
         if (attachment && !attachment.didCancel && !attachment.error) {
             const result = getDataForUpload(attachment);
-            this.setState({result});
-        }
-    }
-
-    /**
-     * Triggers the `onPicked` callback with the selected attachment
-     */
-    completeAttachmentSelection() {
-        if (this.state.result) {
-            this.state.onPicked(this.state.result);
+            this.state.onPicked(result);
         }
     }
```

Comment:
This method is no longer needed.
Previously it would get called when the modal closed and only then you'd see the attachment, hence the issue